src/storage/admin/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,158 @@
+
+    for (PartitionID part : parts) {
+        if (req.get_is_offline()) {
+            LOG(INFO) << "Offline";
+            std::unique_ptr<kvstore::KVIterator> iter;
+            auto prefix = NebulaKeyUtils::prefix(part);
+            auto ret = kvstore_->prefix(space, part, prefix, &iter);
+            if (ret != kvstore::ResultCode::SUCCEEDED) {
+                LOG(ERROR) << "Processing Part " << part << " Failed";
+                this->pushResultCode(to(ret), part);
+                onFinished();
+                return;
+            }
+
+            std::vector<kvstore::KV> data;
+            data.reserve(FLAGS_rebuild_index_batch_size);
+            int32_t batchSize = 0;
+            VertexID currentSrcVertex = -1;
+            VertexID currentDstVertex = -1;
+            while (iter && iter->valid()) {
+                if (batchSize >= FLAGS_rebuild_index_batch_size) {
+                    callingNum_ += 1;
+                    doPut(space, part, std::move(data));
+                    data.clear();
+                    batchSize = 0;
+                }
+
+                auto key = iter->key();
+                auto val = iter->val();
+                if (!NebulaKeyUtils::isEdge(key) ||
+                    NebulaKeyUtils::getEdgeType(key) != edgeType) {
+                    iter->next();
+                    continue;
+                }
+
+                auto source = NebulaKeyUtils::getSrcId(key);
+                auto destination = NebulaKeyUtils::getDstId(key);
+                if (currentSrcVertex == source || currentDstVertex == destination) {
+                    iter->next();
+                    continue;
+                } else {
+                    currentSrcVertex = source;
+                    currentDstVertex = destination;
+                }
+                auto ranking = NebulaKeyUtils::getRank(key);
+                auto reader = RowReader::getEdgePropReader(schemaMan_,
+                                                           std::move(val),
+                                                           space,
+                                                           edgeType);
+                auto values = collectIndexValues(reader.get(), item->get_fields());
+                auto indexKey = NebulaKeyUtils::edgeIndexKey(part, indexID, source,
+                                                             ranking, destination, values);
+                data.emplace_back(std::move(indexKey), "");
+                iter->next();
+            }
+            doPut(space, part, std::move(data));
+        } else {
+            auto atomic = [space, part, edgeType, item, this]() -> std::string {
+                return partitionRebuildIndex(space, part, edgeType, item);
+            };
+
+            auto callback = [space, part, this](kvstore::ResultCode code) {
+                handleAsync(space, part, code);
+            };
+
+            this->kvstore_->asyncAtomicOp(space, part, atomic, callback);
```

Comment:
I think you'd better delete these code, because This is a semi-finished product of online rebuild index. WDYT?

Reply:
if the discard index is not exist or it's just loading data and no write currently, we can build index in batch. Otherwise we should  remove indexes and write the new index in atomic operation.